### PR TITLE
improves deployment

### PR DESCRIPTION
**Pull request by rudolfix**
Reorganizes deployment scripts and provides deployment documentation in README.
Fixes a typo in ACCESS_CONTROLLER role name.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -10,6 +10,13 @@ abbrev@1.0.x:
   version "1.0.9"
   resolved "https://registry.yarnpkg.com/abbrev/-/abbrev-1.0.9.tgz#91b4792588a7738c25f35dd6f63752a2f8776135"
 
+accepts@~1.3.4:
```
Comment (krzkaczor) [inline yarn.lock]: Why is this file changed? there are no changes to package json. Did you updated deps or did something that would otherwise trigger this?

File: migrations/6_relinquish_control.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
+require("babel-register");
+const controlAccounts = require("./accounts").default;
+
+const RoleBasedAccessControl = artifacts.require("RoleBasedAccessControl");
+const EuroToken = artifacts.require("EuroToken");
+
+// Needs to match contracts/AccessControl/RoleBasedAccessControl.sol:TriState
+const TriState = { Unset: 0, Allow: 1, Deny: 2 };
```
Comment (krzkaczor) [inline migrations/6_relinquish_control.js]: this is repeated multiple times. I would suggest extracting it to separate file and exporting it from there.

File: migrations/accounts.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+// Maps roles to accounts on live network
+const ACCESS_CONTROLLER = "0xaa11C97Be40Cdf6e24229EAA731EE3701C3B9493";
+const LOCKED_ACCOUNT_ADMIN = "0xE8Bba2F6bcF6893E66F316d7f2546ee8534F2868";
+const WHITELIST_ADMIN = "0x45FE1297E62c57179BEbe2cdcBcf22b9005769b4";
+const PLATFORM_OPERATOR_WALLET = "0xfE9641B0f229a2D55a3D1aBC77eCa90648BD5357";
+const PLATFORM_OPERATOR_REPRESENTATIVE =
+  "0x11CA6fdC4117feb52d8b81adA7AdEBd519dD8D08";
+const EURT_DEPOSIT_MANAGER = "0x113b51BFAcbF6BC25B553E2a0B1C90741c25C6eB";
+
+export default function controlAccounts(network, accounts) {
```
Comment (krzkaczor) [inline migrations/accounts.js]: I think it would be much wiser to return object here. Then calling it and destructing it is a way more readable and self-describing.

File: migrations/6_relinquish_control.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+// Maps roles to accounts on live network
+let ACCESS_CONTROLLER;
+
+module.exports = function deployContracts(deployer, network, accounts) {
+  // do not deploy testing network
+  if (network === "inprocess_test" || network === "coverage") return;
+  [ACCESS_CONTROLLER] = controlAccounts(network, accounts);
+
+  deployer.then(async () => {
```
Comment (krzkaczor) [inline migrations/6_relinquish_control.js]: why not `await deployer`?

File: migrations/5_amend_agreements.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  if (network.endsWith("_live")) {
+    console.log("---------------------------------------------");
+    console.log(
+      `Must use ${PLATFORM_OPERATOR_REPRESENTATIVE} account to deploy agreements on live newtork`
+    );
+    console.log("---------------------------------------------");
+    return;
+  }
+
+  deployer.then(async () => {
```
Comment (krzkaczor) [inline migrations/5_amend_agreements.js]: why not `await deployer`

File: migrations/5_amend_agreements.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
+require("babel-register");
+const controlAccounts = require("./accounts").default;
+
+const Neumark = artifacts.require("Neumark");
+const Commitment = artifacts.require("Commitment");
+
+// Maps roles to accounts on live network
+let PLATFORM_OPERATOR_REPRESENTATIVE;
+const RESERVATION_AGREEMENT =
+  "ipfs:QmPXME1oRtoT627YKaDPDQ3PwA8tdP9rWuAAweLzqSwAWT";
```
Comment (krzkaczor) [inline migrations/5_amend_agreements.js]: I would take these hardcoded constants and the ones in accounts.js file and put them in separate file so they can be easily changed in future

File: yarn.lock (hunk 1 above)
Comment (rudolfix) [reply yarn.lock]: yes, this is what I did. should we rollback yarn file?

File: migrations/5_amend_agreements.js (hunk 5 above)
Comment (rudolfix) [reply migrations/5_amend_agreements.js]: please nooo! if you export async function from test to truffle, it will fail in so unpredictable way. we spent hours to find this pattern, which works deterministic. better not touch it now!

File: migrations/accounts.js (hunk 3 above)
Comment (rudolfix) [reply migrations/accounts.js]: done!

File: migrations/6_relinquish_control.js (hunk 2 above)
Comment (rudolfix) [reply migrations/6_relinquish_control.js]: done!

File: migrations/5_amend_agreements.js (hunk 5 above)
Comment (krzkaczor) [reply migrations/5_amend_agreements.js]: sure ;D i am fine with this solution if it works reliably

File: yarn.lock (hunk 1 above)
Comment (krzkaczor) [reply yarn.lock]: if it works it's fine ;) I am was just curious why it changed
